bundles/de.uka.ipd.sdq.simucomframework/src/de/uka/ipd/sdq/simucomframework/usage/UserProcessCountMonitor.java

```diff
@@ -0,0 +1,52 @@
+package de.uka.ipd.sdq.simucomframework.usage;
+
+import org.apache.log4j.Logger;
+
+import de.uka.ipd.sdq.simulation.abstractsimengine.ISimProcess;
+import de.uka.ipd.sdq.simulation.abstractsimengine.ISimProcessListener;
+import de.uka.ipd.sdq.simulation.abstractsimengine.ISimulationControl;
+
+public class UserProcessCountMonitor implements IUserProcessMonitor, ISimProcessListener {
+
+    private static final Logger LOGGER = Logger.getLogger(UserProcessCountMonitor.class.getName());
+
+    private final ISimulationControl simulation;
+    private final int maxProcessCount;
+
+    private int processCounter;
+
+    public UserProcessCountMonitor(ISimulationControl simulation, int maxProcessCount) {
+        this.simulation = simulation;
+        this.maxProcessCount = maxProcessCount;
+    }
+
+    @Override
+    public void registerProcess(ISimProcess process) {
+        process.addProcessListener(this);
+    }
+
+    @Override
+    public void notifyStarted(ISimProcess process) {
+        processCounter++;
+        if (processCounter > 0) {
+            if (processCounter >= maxProcessCount) {
+                LOGGER.info(String.format("Reached max user process count %d -> stop simulation", maxProcessCount));
+                simulation.stop();
+            }
+        }
+    }
+
+    @Override
+    public void notifyTerminated(ISimProcess process) {
+        processCounter--;
```

Comment:
Beware, you are running in a multi threaded environment. I would feel more comfortable if you'd use something synchronized, as e. g. an AtomicInteger for the processCounter variable.